drivers/xbee/xbee.c

```diff
@@ -603,6 +603,8 @@ int xbee_init(netdev_t *dev)
     _at_cmd(xbee, "ATAP1\r");
     /* apply AT commands */
     _at_cmd(xbee, "ATAC\r");
+    /* disable xbee CTS, on arduino boards since it is connected to the reset pin */
+    _at_cmd(xbee, "ATD7 0\r");
```

Comment:
You are disabling CTS always, is it possible that some UART implementation is using Flow control and may need this?

Reply:
You are right, I didn't because we don't use flow control in our code AFAIK, and because I was lazy. But it should be optional, I'll try to do it.

Reply:
I think this change would address your comment:

```
#ifndef MODULE_PERIPH_UART_HW_FC
    /* disable xbee CTS and RTS, unless hardware flow control is used */
    _at_cmd(xbee, "ATD7 0\r");
    _at_cmd(xbee, "ATD6 0\r");
#endif
```

Going back to when this fix is needed:

- `xbee` is used with an `arduino` `BOARD` with `ICSP` connector
- `xbee` is used on an `arduino` `shield` which connects to the `ICSP` connecot

There is no way of checking at compile time what kind of `arduino-shield` is being used or if the `BOARD` has an ICSP connector. So I'm thinking this should be addressed with a warning, I would maybe have something like this:

```
#ifndef MODULE_PERIPH_UART_HW_FC
    /* disable xbee CTS and RTS, unless hardware flow control is used */
     DEBUG("[xbee] init: WARNING when using arduino boards and arduino xbee " \
           "shield with ICSP connector, HW flow control can't be used\n");
    _at_cmd(xbee, "ATD7 0\r");
    _at_cmd(xbee, "ATD6 0\r");
#endif
```

We could also add `HAS_ICSP` to all `arduino-boards`, and if `HAS_ICSP`, hardware flow control would be disabled by default for those boards, or a `FEATURES_PROVIDED += board_arduino` `FEATURES_REQUIRED += board_arduino`. But this would be a bigger change, and if we want this in the release I would rather keep it out.

Reply:
I'd say let's go for now with the warning message and checking for `MODULE_PERIPH_UART_HW_FC` (BTW this can be done using a C conditional and `IS_USED`), so we can make it to RC2. And improve the solution on a follow-up PR.